EventEnabledInsurance/prereqs.sh

```diff
@@ -158,14 +222,184 @@ stringData:
+
+# create common roles for tasks
+echo "INFO: Create common roles for tasks"
+if cat $CURRENT_DIR/../CommonPipelineResources/cicd-roles.yaml |
+  sed "s#{{NAMESPACE}}#$namespace#g;" |
+  oc apply -f -; then
+   echo -e "\n$tick INFO: Successfully created roles for tasks in the '$namespace' namespace"
+else
+  echo -e "\n$cross ERROR: Failed to create roles for tasks in the '$namespace' namespace"
+fi
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+# create common role bindings for roles
+echo "INFO: Create role bindings for roles"
+if cat $CURRENT_DIR/../CommonPipelineResources/cicd-rolebindings.yaml |
+  sed "s#{{NAMESPACE}}#$namespace#g;" |
+  oc apply -f -; then
+    echo -e "\n$tick INFO: Successfully applied role bindings for roles in the '$namespace' namespace"
+else
+  echo -e "\n$cross ERROR: Failed to apply role bindings for roles in the '$namespace' namespace"
+fi #cicd-rolebindings.yaml
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+echo "INFO: Creating roles for creating deployment in the '$namespace' namespace for simulator app"
+if oc apply -f $CURRENT_DIR/QuoteLifecycleSimulator/simulator-pipeline-resources/simulator-roles.yaml; then
+    echo -e "\n$tick INFO: Successfully created roles for creaating deployment in the '$namespace' namespace for simulator app"
+else
+  echo -e "\n$cross ERROR: Failed to create roles for creaating deployment in the '$namespace' namespace for simulator app"
+  exit 1
+fi #simulator-roles.yaml
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+echo "INFO: Creating rolebinding for creating deployment in the '$namespace' namespace for creating simulator app"
+if cat $CURRENT_DIR/QuoteLifecycleSimulator/simulator-pipeline-resources/simulator-rolebindings.yaml |
+  sed "s#{{NAMESPACE}}#$namespace#g;" |
+  oc apply -f -; then
+    echo -e "\n$tick INFO: Successfully created rolebinding for creating deployment in the '$namespace' namespace for simulator app"
+else
+  echo -e "\n$cross ERROR: Failed to create rolebinding for creating deployment in the '$namespace' namespace for simulator app"
+  exit 1
+fi #simulator-rolebindings.yaml
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+time=0
+echo "INFO: Waiting for upto 10 minutes for git-clone cluster task to be available before creating the pipeline and the pipelinerun..."
+while [ "$($TKN clustertask ls | grep git-clone | awk '{print $1}')" != "git-clone" ]; do
```

Comment:
Why not just check if `oc get clustertask git-clone` succeeds? Saves messing around with grep and awk, and is safer. If there's a cluster task with a string that contains `git-clone` (I.e. `my-git-clone-task`), then the grep/awk won't work.

Reply:
I was wondering that would the cluster task be named anything else at all apart from git-clone considering that it is created by the ocp pipelines?

Reply:
Done

```
INFO: Waiting for upto 10 minutes for git-clone cluster task to be available before creating the pipeline and the pipelinerun...
Error from server (NotFound): clustertasks.tekton.dev "git-clones" not found
git-clone                                22 minutes ago

INFO: The cluster task ''git-clone' is not yet available, waiting for upto 10 minutes. Waited 0 minute(s).

INFO: Cluster task 'git-clone' is now available

git-clone                                23 minutes ago

----------------------------------------------------------------------------------------------------------------------------------------------------------

INFO: Creating the pipeline to build and deploy the simulator app in 'cp4i1' namespace
pipeline.tekton.dev/eei-simulator-pipeline configured

✅ INFO: Successfully applied the pipeline to build and deploy the simulator app in in 'cp4i1' namespace

----------------------------------------------------------------------------------------------------------------------------------------------------------
```